libraries/common/map.effekt

```diff
@@ -0,0 +1,714 @@
+module map
```

Comment:
In https://github.com/effekt-lang/effekt/pull/488, we agreed to use constructors like `$collection_name` for `array`, `ref`, and `bytes`. For example:
```scala
// Creates a new Array of size `size` filled with the value `init`
extern global def array[T](size: Int, init: T): Array[T]
 
// Allocates new bytes with the given `capacity`, setting its values to `0`.
extern io def bytes(capacity: Int): Bytes

// Creates a new reference with the initial value `init`.
extern global def ref[T](init: T): Ref[T]
 ```

There are two main ways to make a `map` / `set`:
- empty `map`/`set`
- `map`/`set` created from a (possibly unordered) list

Does the convention mention above apply anywhere here? Personally, I'd always use these namespaced: `map::empty()` and `map::fromList([...])`, but I'm interested in what y'all think.